### PR TITLE
Adding JWT user mapping

### PR DESCRIPTION
**Pull request by erickduran**
- Adding dynamic import via `ACL_MODULE` to check for appropriate access
- Response JWT will replace the `user` with the provided `id` if the mapping allows it, otherwise will return 403
- Response JWT will include `parent` (the original authenticated username) moving forward

### CODE REVIEW

File: confidant/routes/jwks.py [hunk 1]
```diff
@@ -6,11 +6,14 @@
...
 from confidant.schema.jwks import jwt_response_schema, JWTResponse, \
     jwks_list_response_schema, JWKSListResponse
-
+from confidant.settings import JWT_MAPPING_MODULE
+from confidant.utils import misc
 
 logger = logging.getLogger(__name__)
 blueprint = blueprints.Blueprint('jwks', __name__)
 
+jwt_mapping_check = misc.load_module(JWT_MAPPING_MODULE)
```
Comment (skiptomyliu) [inline confidant/routes/jwks.py:15]: let's conform to use `acl_module_check = misc.load_module(settings.ACL_MODULE)` instead.  This centralizes the RBAC logic.

File: confidant/routes/jwks.py [hunk 2]
```diff
@@ -39,16 +42,29 @@ def get_token():
...
     if not environment:
         return jsonify({'error': 'Please specify an environment'}), 400
 
     payload = {
-        'user': user,
         'is_service': authnz.user_is_service(user),
+        'parent': parent,
     }
+    target_user = request.args.get('target_user', type=str)
+    if target_user and JWT_MAPPING_MODULE:
```
Comment (skiptomyliu) [inline confidant/routes/jwks.py:57]: let's change this to

```bash
if not acl_module_check(resource_type='jwt', action='get', resource_id=id):
        msg = f"{authnz.get_logged_in_user()} does not have access to get JWT {id}"
        error_msg = {'error': msg}
        return jsonify(error_msg), 403
```

File: confidant/routes/jwks.py [hunk 3]
```diff
@@ -6,11 +6,14 @@
...
+from confidant.utils import misc
 
 logger = logging.getLogger(__name__)
 blueprint = blueprints.Blueprint('jwks', __name__)
 
+jwt_mapping_check = misc.load_module(JWT_MAPPING_MODULE)
+
 
 @blueprint.route('/v1/jwks/token', methods=['GET'])
 @authnz.require_auth
```
Comment (skiptomyliu) [inline confidant/routes/jwks.py:19]: we need to change this to

```python
def get_token(id):
```

the `id` will the be the token (resource_id) we're attempting to request.

File: confidant/routes/jwks.py [hunk 4]
```diff
@@ -39,16 +42,23 @@ def get_token():
     :statuscode 200: Success
     :statuscode 400: JWTs are not supported for this user
     """
-    user = authnz.get_logged_in_user()
+    parent = authnz.get_logged_in_user()
```
Comment (skiptomyliu) [inline confidant/routes/jwks.py:45]: should we change this to `host`?  trying to think of a better name 🤷

Comment (skiptomyliu) [reply confidant/routes/jwks.py:45]: eh, should be fine for now.

**Review by skiptomyliu — DISMISSED** [review]

**Review by skiptomyliu — APPROVED** [review]

File: confidant/routes/jwks.py (hunk 3 above)
Comment (Yueren-Wang) [reply confidant/routes/jwks.py:19]: so this id will be the envoy_service name?

Comment (skiptomyliu) [reply confidant/routes/jwks.py:19]: @Yueren-Wang yes

**Pull request merged**